File: C++/String/3_duplicate_character.cpp
/*
Idea:
using map to store count of each element , if count exceeds 1, print the answer.
*/

#include <bits/stdc++.h>
using namespace std;

void printDups(string str)
{
    unordered_map<char, int> mp;
    for (int i = 0;i < str.size();++i)
        mp[str[i]]++;

    for (auto it : mp)
        if (it.second > 1)
            cout << it.first << " ";
    cout << "\n";
}


int main()
{
    string str = "test string";
    printDups(str);
    return 0;
}

File: Python/String/3_duplicate_character.py
"""
Idea:
using dictionary to store count of each element , if count exceeds 1, print the answer.dea:
"""


def printDups(Str):
    dic = {}
    for i in range(len(Str)):
        if Str[i] in dic:
            dic[Str[i]] += 1
        else:
            dic[Str[i]] = 1

    for i, j in dic.items():
        if j > 1:
            print(i, end=' ')


if __name__ == '__main__':
    T = int(input())
    for i in range(T):
        S = input()
        print(printDups(S))
